### ISSUE TITLE
Getting Error "Unknown identifier ClassDefEx-1.$.startDate" when dynamically loading Data.

### ISSUE THREAD

**Issue by jsakalos** [bryntum/support#5322]
The poster triggered this error in Angular but it is also reproducible in vanilla. Steps to reproduce:

1. In `SchedulerPro/examples/dependencies/ replace the content of `app.js` with the following:
```
import '../_shared/shared.js'; // not required, our example styling etc.
import SchedulerPro from '../../lib/SchedulerPro/view/SchedulerPro.js';
import '../../lib/Scheduler/feature/ResourceTimeRanges.js';
import '../../lib/Grid/feature/Tree.js';
import Toast from '../../lib/Core/widget/Toast.js';
import StringHelper from '../../lib/Core/helper/StringHelper.js';

const scheduler = new SchedulerPro({
    // A Project holds the data and the calculation engine for Scheduler Pro. It also acts as a CrudManager, allowing
    // loading data into all stores at once
    project : {
        eventStore : {
            fields : ['highlight']
        }
    },

    appendTo              : 'container',
    startDate             : '2020-03-23',
    endDate               : '2020-03-24',
    rowHeight             : 60,
    barMargin             : 15,
    eventStyle            : 'colored',
    viewPreset            : 'hourAndDay',
    highlightSuccessors   : true,
    highlightPredecessors : true,

    features : {
        dependencies : {
            highlightDependenciesOnEventHover : true
        },
        dependencyEdit : {
            // Allow editing lag in the dependency editor
            showLagField : true
        },
        eventDrag : {
            // Prevent reassigning events using drag and drop
            constrainDragToResource : true
        },
        resourceTimeRanges : true,

        tree : true
    },

    columns : [
        // A column using a custom render to display an icon + text
        {
            text  : 'Resource',
            width : 150,
            field : 'name',
            type  : 'tree'
        }
    ],

    eventRenderer({ eventRecord, renderData }) {
        // Add 'highlight' css class for highlighted records
        renderData.cls.highlight = eventRecord.highlight;

        return StringHelper.encodeHtml(eventRecord.name);
    },

    tbar : [
        {
            type    : 'checkbox',
            label   : 'Highlight dependent events',
            checked : true,
            onChange({ checked }) {
                if (checked && !scheduler.selectedEvents.length) {
                    Toast.show('Select an event to highlight the dependency chain');
                }

                scheduler.highlightSuccessors = scheduler.highlightPredecessors = checked;
            }
        }
    ]
});

const project = scheduler.project;

setInterval(() => {
    console.log('setting data');
    project.resources = [{ id : '1', expanded : true, children : { id : '2', name : 'test' }, dbid : 7, name : 'Test' }];
    project.events = [
        { id : '1', startDate : '2021-09-20T00:00:00', endDate : '2021-10-08T12:00:00', resourceId : '2', name : 'Test Event' },
        { id : '2', startDate : '2021-09-20T00:00:00', endDate : '2021-10-08T12:00:00', resourceId : '2', name : 'Test Event' }
    ];
    project.dependencies = [{ from : '1', to : '2' }];
}, 100);
```
2. Run in the browser and change to another tab in the browser, then back. It crashes:

![set-data-bug](https://user-images.githubusercontent.com/204844/192330371-08f29cd4-7417-49cd-b913-dada829ffc39.gif)

[Forum post](https://www.bryntum.com/forum/viewtopic.php?f=44&t=22448&p=111049#p111049)

Hey,
I am trying to load data on "visibledaterangechange"- event. But after reloading the data by using the slider it seems like an internal error occurred.

```
ERROR Error: Uncaught (in promise): Error: Unknown identifier ClassDefEx-1.$.startDate
Error: Unknown identifier ClassDefEx-1.$.startDate
    at throwUnknownIdentifier (schedulerpro.module.js:188861:9)
    at EngineTransaction.addEdge (schedulerpro.module.js:189706:27)
    at EngineTransaction.onReadIdentifier (schedulerpro.module.js:189765:33)
    at EngineTransaction.calculateTransitionsStackGen (schedulerpro.module.js:189920:47)
    at calculateTransitionsStackGen.next (<anonymous>)
    at EngineTransaction.calculateTransitions (schedulerpro.module.js:189814:19)
    at calculateTransitions.next (<anonymous>)
    at schedulerpro.module.js:188232:112
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at resolvePromise (zone.js:1262:1)
```

To reproduce this error, I have attached a dummy Angular component, where you just need to use the slider "fast".

Is there something I'm doing wrong or should the data be loaded differently?

**Comment by canonic-epicure** [comment]
Found another bug, while examining this one - selecting an event after the manual data refresh (w/o setInterval) throws an exception. Seems to be caused by the `repopulateOnDataset` option

**Comment by canonic-epicure** [comment]
Extracted to #5352

**Comment by EricBorland** [comment]
Hi there, I've downloaded the 5.2.1 version right now (for react) and yet I still see this issue, is the fix really included in 5.2.1?
